components/atoms/Spin/Spin.tsx

```diff
@@ -0,0 +1,71 @@
+import * as Stitches from "@stitches/react";
+
+import { styled, keyframes } from "../../../stitches.config";
+import { colorVariants } from "../../utils/color-variants";
+
+const loading = keyframes({
```

Comment:
i think this can also include a timeout for loading, i mean for example it will rotate for 5 mins and then will give out some error.